src/provider.rs

```diff
@@ -19,6 +19,17 @@ impl<'a> FolderRegionProvider<'a> {
 
         FolderRegionProvider { folder_path }
     }
+
+    // leave implementing this to the specific provider,
+    // makes function declaration bearable for now
+    pub fn iter(&self) -> Result<impl Iterator<Item=RegionPosition>, io::Error> {
```

Comment:
Return `Region`?

Reply:
I intentionally don't do this, because loading the Region takes way more time to load (theoretically) and it forces a user to load all regions. 
The way it is right now you can filter the Positions before hitting the disk again, which is useful if you only want to load regions in say some bounds.

What we could do is rename this method to something like `iter_positions` and make a `iter_regions` that yields them directly.

Reply:
I renamed `iter` to `iter_positions` for now, but before I hold the fixes up til i get iter_regions to work, I'll probably make another PR when I get that sorted out.